pkg/component/component_test.go

```diff
@@ -2214,7 +2216,8 @@ func TestToComponents(t *testing.T) {
 				},
 			},
 			headers: &testHeadersProvider{headers: map[string]string{
-				"header-one": "val-1",
+				"header-one":   "val-1",
+				"header-three": "val-3-diff",
```

Comment:
Just making sure I understand the precedence order here correctly: this value of `header-three`, `val-3-diff` gets evaluated second, and therefore it should **not** be used since `header-three` was already set to `val-3` first, correct?